src/main/java/uk/gov/justice/hmpps/prison/service/OffenderAssessmentService.java

```diff
@@ -120,20 +124,39 @@ private AssessmentQuestion getAssessmentQuestionAndAnswers(final AssessmentEntry
             answers.stream().skip(1).collect(toList()));
     }
 
-    public List<AssessmentClassification> getOffendersAssessmentRatings(List<String> offenderList) {
-        return offenderList.stream().map(this::getOffenderCurrentAssessmentRating).collect(Collectors.toList());
+    @Transactional(readOnly = true)
+    public List<AssessmentClassification> getOffendersAssessmentRatings(final List<String> offenderNos) {
+        final var classifications = new ArrayList<AssessmentClassification>();
+        final var batch = Lists.partition(new ArrayList<>(offenderNos), maxBatchSize);
+        batch.forEach(offenderNosBatch -> classifications.addAll(getOffendersCurrentAssessmentRating(offenderNosBatch)));
+        return classifications;
+    }
+
+    private List<AssessmentClassification> getOffendersCurrentAssessmentRating(final List<String> offenderNos) {
+        final var assessmentsLatestFirstForAllOffenders = repository.findByCsraAssessmentAndByOffenderNos_OrderByLatestFirst(offenderNos);
+        final var assessmentsLatestFirstByOffenderNo = assessmentsLatestFirstForAllOffenders.stream().collect(groupingBy(OffenderAssessment::getOffenderNo));
+        return assessmentsLatestFirstByOffenderNo.entrySet().stream()
+            .map(e -> getOffenderCurrentAssessmentRating(e.getKey(), e.getValue()))
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .collect(toList());
     }
 
-    private AssessmentClassification getOffenderCurrentAssessmentRating(String offenderNo) {
-        var currentClassification = getCurrentCsraClassification(offenderNo);
+    private Optional<AssessmentClassification> getOffenderCurrentAssessmentRating(final String offenderNo, final List<OffenderAssessment> assessmentsLatestFirst) {
+        var currentClassification = calculateCurrentCsraClassification(assessmentsLatestFirst);
         if (currentClassification == null) {
-            return AssessmentClassification.builder().offenderNo(offenderNo).build();
+            return Optional.empty();
         }
-        return AssessmentClassification.builder()
+        return Optional.of(AssessmentClassification.builder()
             .offenderNo(offenderNo)
             .classificationCode(currentClassification.classificationCode)
             .classificationDate(currentClassification.classificationDate)
-            .build();
+            .build());
+    }
+
+    private CurrentCsraAssessment calculateCurrentCsraClassification(final List<OffenderAssessment> offenderAssessmentsLatestFirst) {
```

Comment:
Would be be better to return a Optional here?

Reply:
I spent about 5 minutes thinking about this and decided to leave as is. The other method that does use Optional was done so we could better chain calls in a stream.